src/main/java/org/reactome/release/qa/check/ChimericInstancesCheck.java

```diff
@@ -32,6 +32,7 @@
 @SliceQACheck
 public class ChimericInstancesCheck extends AbstractQACheck {
 
+    private static final String skiplist = "resources/chimerism_reference_constraint_violations.txt";
```

Comment:
Should this really be hardcoded? Or could the path to a file be passed in as an argument? Maybe some skip-list related functionality could be added to AbstractQACheck.

Reply:
UPDATE: I read further and can see there _is_ common skip list functionality in AbstractQACheck. Could the _name_ of the skip list file be stored in a configuration specific to the QA Check, and then read by AbstractQACheck so that the specific QA Checks don't need to know such specific information about it? For example, if there was a file `chimericInstancesCheck.properties`, containing `skipList=resources/chimerism_reference_constraint_violations.txt` then AbstractQACheck could automatically look for the skipList property, and load it if it's found.

Reply:
Or maybe one config file with all the skips lists, not sure what would be better right now...

Reply:
Oh true, I was unaware of that functionality. I'll look into refactoring this code to use the existing system, thanks for catching that

Reply:
OK, so I've looked into this and from what I've looked at, I don't know that the proposed approach is necessarily better.

As I understand it, @SolomonShorser-OICR , you looked at `AbstractQACheck.getConfigurationFile`, which does in fact have functionality for automatically reading in files. So in theory, you do it the following way:

1) Call `getConfigurationFile` for the step, which would look for `resources/StepName.txt` which would contain the property `skiplist=resources/step_name_skip_list.txt`, that points to the actual skiplist file.
2) Use the same methods to read in the file as before, adding them to a skiplist.

Really, what we would be doing is removing the hard-coded skiplist file name, but doubling the amount of files per 'skiplist' class in the resources folder. Thats not necessarily a dealbreaker, but I think that in comparing the two methods (step-specific config file pointing to skiplist file vs. hard-coded filename pointing to step-specific skiplist file), an argument for the hard-coded filename is that it would be easier to understand how the system works, while an argument for the config file is that it would make it better prepared for future 'step-specific' configurations. 

My inclination is to keep the system as is, since I don't see a very strong advantage to one over the other, except that one has already been completed! Happy to discuss further

Reply:
> ...doubling the amount of files per 'skiplist' class in the resources folder...

I'm confused about this.
1) What is a "'skiplist' class"? There is no `SkipList` class, at least not that I'm aware of. The skip list is a list read in from a file.
2) Why would the number of files double? If there's already an existing configuration file, wouldn't the path to the skip list just be stored in there?

Reply:
1. Well coincidentally Joel proposed the literal SkipList class down below, I meant it more informally, as 'a class that utilizes a skiplist'. 
2. The number of files double because _there _isn't_ already an existing configuration file - just a skiplist file that is hard-coded in the classes that need one. So the proposal above would involve creating a `StepName.txt` file (the `getConfigurationFile` method assumes a txt suffix), which would be used to point to the actual skip file.

Reply:
The path to the skiplist could also be passed in to the program as an argument.

Reply:
Also, I somehow can't seem to find where @jweiser talks about creating a new SkipList class. Could you link to it?

Reply:
https://github.com/reactome/release-qa/pull/27#pullrequestreview-526785941

Reply:
Ah, there it is. I like that idea!

Reply:
> The path to the skiplist could also be passed in to the program as an argument.

The way that these QA tests are run currently, during weekly QA, is as an 'all or nothing' program - There is a config file that specifies that @SliceQACheck, @GraphQACheck, and @DiagramQACheck tests will be run, and then by their association with any of those three 'checks', a QA test is run.

This is to say that an implicit skiplist is cleaner for this form of execution. If we are very against a hard-coded file name, I will look at other solutions, but I'm not sure I understand the dislike of this approach (not a complaint, just curious)

Reply:
https://github.com/reactome/release-qa/pull/27#discussion_r533553913 > Fair enough, but the issue of hard-coded vs. not hard-coded skiplist file names still exists with a `SkipList` class